static/app/views/issueList/searchBar.spec.tsx

```diff
@@ -238,4 +236,44 @@ describe('IssueListSearchBar', function () {
       expect(await screen.findByRole('option', {name: 'someTag'})).toBeInTheDocument();
     });
   });
+
+  describe('Tag Values', function () {
+    const {router: routerWithFlag, organization: orgWithFlag} = initializeOrg();
+    orgWithFlag.features = ['issue-stream-search-query-builder'];
+
+    const newDefaultProps = {
+      organization: orgWithFlag,
+      query: '',
+      statsPeriod: '7d',
+      onSearch: jest.fn(),
+    };
+
+    it('displays the correct tag values for a key', async () => {
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/tags/',
+        method: 'GET',
+        body: [],
+      });
+      const tagKey = 'random';
+      const tagValueMock = MockApiClient.addMockResponse({
+        url: `/organizations/org-slug/tags/${tagKey}/values/`,
+        method: 'GET',
+        body: [],
+      });
+
+      render(<IssueListSearchBar {...newDefaultProps} />, {
+        router: routerWithFlag,
+      });
+
+      await userEvent.click(screen.getByRole('combobox', {name: 'Add a search term'}));
+      await userEvent.paste(tagKey, {delay: null});
+      await userEvent.click(screen.getByRole('row', {name: tagKey}));
+
+      waitFor(() => {
+        // Expected twice since we make one request for values in events dataset
+        // and another for values in IssuePlatform dataset.
+        expect(tagValueMock).toHaveBeenCalledTimes(2);
```

Comment:
Might make sense to return a tag from each mock response and see that they are actually rendered

Reply:
@MichaelSun48 looks like the test got fixed a bit, but still missing this at the end

`expect(await screen.findByRole('option', { name: 'randomValue' })).toBeInTheDocument()`

Reply:
@malwilley 
Unless I'm misunderstanding, I believe this is done on line 291?

Reply:
Oops. I misread `tagValue` as `tagKey`. Looks good!